src/dune_lang/package_info.ml

```diff
@@ -109,6 +109,70 @@ let encode_fields
     ]
 ;;
 
+let valid_maintenance_intent =
+  let open Decoder in
+  map_validate (located string) ~f:(fun (loc, str) ->
+    let rec loop i =
```

Comment:
I first thought that `loop` would be the entry point, but this is not the case. I think in such case it would be nicer if `loop` had a more descriptive name to explain what it does in the parsing.